src/hooks/__internal__/useFloating/useFloating.test.tsx

```diff
@@ -41,120 +41,100 @@ const MockComponent = ({
+  expect(computePositionSpy.mock.calls[0][0]).toBe(
+    screen.getByTestId("reference-element")
+  );
+  expect(computePositionSpy.mock.calls[0][1]).toBe(
+    screen.getByTestId("floating-element")
+  );
+  expect(computePositionSpy.mock.calls[0][2]).toMatchObject({
+    strategy: "fixed",
+    placement: "top",
   });
+});
 
-  it("invokes autoUpdate with proper arguments", () => {
-    const autoUpdateSpy = jest.spyOn(floatingUi, "autoUpdate");
+test("autoUpdate is invoked with proper arguments across different states", () => {
+  const autoUpdateSpy = jest.spyOn(floatingUi, "autoUpdate");
 
-    const { rerender } = render(<MockComponent isOpen />);
+  const { rerender } = render(<MockComponent isOpen />);
 
-    expect(autoUpdateSpy.mock.calls[0][0]).toBe(
-      screen.getByTestId("reference-element")
-    );
-    expect(autoUpdateSpy.mock.calls[0][1]).toBe(
-      screen.getByTestId("floating-element")
-    );
-    expect(autoUpdateSpy.mock.calls[0][3]).toMatchObject({
-      animationFrame: undefined,
-    });
-
-    rerender(<MockComponent isOpen animationFrame />);
-    expect(autoUpdateSpy.mock.calls[1][3]).toMatchObject({
-      animationFrame: true,
-    });
+  expect(autoUpdateSpy.mock.calls[0][0]).toBe(
+    screen.getByTestId("reference-element")
+  );
+  expect(autoUpdateSpy.mock.calls[0][1]).toBe(
+    screen.getByTestId("floating-element")
+  );
+  expect(autoUpdateSpy.mock.calls[0][3]).toMatchObject({
+    animationFrame: undefined,
   });
 
-  it("saves floating element original styles and restores them after closing", async () => {
-    const { rerender } = render(<MockComponent isOpen />);
-
-    await waitFor(() => {
-      const positionedStyle = window.getComputedStyle(
-        screen.getByTestId("floating-element")
-      );
-      expect(positionedStyle).toMatchObject({
-        position: "absolute",
-        top: "0px",
-        left: "0px",
-      });
-    });
-
-    rerender(<MockComponent isOpen={false} />);
-    const originalStyle = window.getComputedStyle(
-      screen.getByTestId("floating-element")
-    );
-
-    expect(originalStyle).toMatchObject({
-      position: "static",
-      top: "100px",
-      left: "50px",
-    });
+  rerender(<MockComponent isOpen animationFrame />);
+  expect(autoUpdateSpy.mock.calls[1][3]).toMatchObject({
+    animationFrame: true,
   });
+});
 
-  it("when using size middleware, the original width and height are restored after closing", async () => {
-    const middleWare = [
-      floatingUi.size({
-        apply({ rects, elements }) {
-          elements.floating.style.height = `${rects.reference.height}px`;
-          elements.floating.style.width = `${rects.reference.width}px`;
-        },
-      }),
-    ];
-
-    const { rerender } = render(
-      <MockComponent isOpen middleware={middleWare} />
-    );
+test("saves floating element original styles and restores them after closing", async () => {
+  const { rerender } = await render(<MockComponent isOpen />);
```

Comment:
**nitpick**: it looks like the await on `render` might be unnecessary. Removing it could simplify the code a bit :)